File: Data Structures & Algorithms/is-anagram/submission-1.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()){
            return false;
        }
        int count[26] = {0};
        for(int i = 0 ; i < s.length() ;i++){
            count[s[i] - 'a']++;
        }
        for(int j = 0 ;j < t.length();j++){
            count[t[j]-'a']--;
        }
        for(int i = 0 ; i < 26;i++){
            if(count[i] != 0){
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> seen = new HashMap<>();
        for(int i : nums){
            if(seen.containsKey(i)){
                int n = seen.get(i);
                seen.put(i, n +1);
            }
            else{
                seen.put(i,0);
            }
        }
        ArrayList<Integer> count = new ArrayList<>();
        for(int i : seen.keySet()){
            count.add(i);
        }
        Collections.sort(count, (a, b) -> seen.get(b) - seen.get(a));
        int[] result = new int[k];
        for(int i = 0 ; i < k ;i++){
            result[i] = count.get(i);
        }
        return result;
    }
}
